core/shared/platform/zephyr/zephyr_file.c

```diff
@@ -0,0 +1,990 @@
+}
+
+__wasi_errno_t
+os_fallocate(os_file_handle handle, __wasi_filesize_t offset,
+             __wasi_filesize_t length)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_ftruncate(os_file_handle handle, __wasi_filesize_t size)
+{
+    struct zephyr_fs_desc *ptr = NULL;
+
+    GET_FILE_SYSTEM_DESCRIPTOR(handle->fd, ptr);
+
+    int rc = fs_truncate(&ptr->file, (off_t)size);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    return __WASI_ESUCCESS;
+}
+
+__wasi_errno_t
+os_futimens(os_file_handle handle, __wasi_timestamp_t access_time,
+            __wasi_timestamp_t modification_time, __wasi_fstflags_t fstflags)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_utimensat(os_file_handle handle, const char *path,
+             __wasi_timestamp_t access_time,
+             __wasi_timestamp_t modification_time, __wasi_fstflags_t fstflags,
+             __wasi_lookupflags_t lookup_flags)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_readlinkat(os_file_handle handle, const char *path, char *buf,
+              size_t bufsize, size_t *nread)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_linkat(os_file_handle from_handle, const char *from_path,
+          os_file_handle to_handle, const char *to_path,
+          __wasi_lookupflags_t lookup_flags)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_symlinkat(const char *old_path, os_file_handle handle, const char *new_path)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_mkdirat(os_file_handle handle, const char *path)
+{
+    struct zephyr_fs_desc *ptr = NULL;
+    int index, rc;
+    char abs_path[MAX_FILE_NAME + 1];
+
+    if (handle == NULL) {
+        return __WASI_EINVAL; // Or another appropriate error code
+    }
+
+    snprintf(abs_path, MAX_FILE_NAME, "%s/%s", prestat_dir, path);
+
+    rc = fs_mkdir(abs_path);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    ptr = zephyr_fs_alloc_obj(true, abs_path, &index);
+    if (!ptr) {
+        return __WASI_EMFILE;
+    }
+    fs_dir_t_init(&ptr->dir);
+
+    handle->fd = index;
+    handle->is_sock = false;
+
+    return __WASI_ESUCCESS;
+}
+
+__wasi_errno_t
+os_renameat(os_file_handle old_handle, const char *old_path,
+            os_file_handle new_handle, const char *new_path)
+{
+    /* `old_handle` need to be the the fd of the file to rename.
+     * `new_handle` will not be used.
+     * paths need to be absolute, no relative path will be accepted.
+     */
+    struct zephyr_fs_desc *ptr = NULL;
+    (void)new_handle;
+
+    char abs_old_path[MAX_FILE_NAME + 1];
+    char abs_new_path[MAX_FILE_NAME + 1];
+
+    snprintf(abs_old_path, MAX_FILE_NAME, "%s/%s", prestat_dir, old_path);
+    snprintf(abs_new_path, MAX_FILE_NAME, "%s/%s", prestat_dir, new_path);
+
+    int rc = fs_rename(abs_old_path, abs_new_path);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    GET_FILE_SYSTEM_DESCRIPTOR(old_handle->fd, ptr);
+
+    ptr->path = strdup(new_path);
```

Comment:
Should check the return value, and should rename the file to old path if strdup failed?